### ISSUE TITLE
twisted.test.test_paths.FilePathTestCase.test_changed relies on reference counting to close a file

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4485]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4485|
|Type|defect|
|Created|2010-06-04 13:08:04Z|
|Branch|https://github.com/twisted/twisted/tree/filepath-test-bug-4485|

```
open(fp.path, 'wb').write("12345678")
```

This requires CPython's reference counting semantics to flush and close the file.  The rest of the test assumes that this will happen immediately after the write call, but it may not.

<details><summary>Searchable metadata</summary>

```
trac-id__4485 4485
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__Twisted_10_1 Twisted-10.1
branch__branches_filepath_test_bug_4485 branches/filepath-test-bug-4485
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__tests tests
time__1275656884000000 1275656884000000
changetime__1275793395000000 1275793395000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [29178]) Branching to 'filepath-test-bug-4485'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [29179]) Explicitly close the file

refs [#4485](https://github.com/twisted/twisted/issues/4485)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli set owner to @exarkun|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [29209]) Merge filepath-test-bug-4485 

Author: exarkun
Reviewer: cyli
Fixes: [#4485](https://github.com/twisted/twisted/issues/4485)

Explicitly close the file in the test for `FilePath.changed` to ensure that
the changes are committed to the filesystem immediately, rather than relying
on the CPython reference counting scheme to collect the file object and force
this to happen in a timely manner.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
